File: spec/javascript/faye_spec.js
JS.ENV.FayeSpec = JS.Test.describe("Faye", function() { with(this) {
  include(JS.Test.Helpers)

  describe("random", function() { with(this) {
    it("returns a 160-bit random number in base 36", function() { with(this) {
      assertMatch( /^[a-z0-9]+$/, Faye.random() )
    }})

    it("always produces the same length of string", function() { with(this) {
      var ids = $R(1,100).map(function() { return Faye.random().length })
      var expected = $R(1,100).map(function() { return 31 })
      assertEqual( expected, ids )
    }})
  }})

  describe("copyObject", function() { with(this) {
    before(function() { with(this) {
      this.object = {foo: "bar", qux: 42, hey: null, obj: {bar: 67}}
    }})

    it("returns an equal object", function() { with(this) {
      assertEqual( {foo: "bar", qux: 42, hey: null, obj: {bar: 67}},
                   Faye.copyObject(object) )
    }})

    it("does not return the same object", function() { with(this) {
      assertNotSame( object, Faye.copyObject(object) )
    }})

    it("performs a deep clone", function() { with(this) {
      assertNotSame( object.obj, Faye.copyObject(object).obj )
    }})
  }})
}})

File: spec/ruby/faye_spec.rb
require "spec_helper"

describe Faye do
  describe :random do
    it "returns a 160-bit random number in base 36" do
      Faye.random.should =~ /^[a-z0-9]+$/
    end

    it "always produces the same length of string" do
      ids = (1..100).map { Faye.random }
      ids.should be_all { |id| id.size == 31 }
    end
  end

  describe :copy_obect do
    let(:object) { {"foo" => "bar", "qux" => 42, "hey" => nil, "obj" => {"bar" => 67}} }

    it "returns an equal object" do
      Faye.copy_object(object).should == {"foo" => "bar", "qux" => 42, "hey" => nil, "obj" => {"bar" => 67}}
    end

    it "does not return the same object" do
      Faye.copy_object(object).should_not be_equal(object)
    end

    it "performs a deep clone" do
      Faye.copy_object(object)["obj"].should_not be_equal(object["obj"])
    end
  end
end
